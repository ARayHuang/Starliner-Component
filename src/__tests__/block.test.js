import React from 'react';
import { shallow, mount } from 'enzyme';
import Block from '../components/block';

describe('block', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-block-className',
			variant: Block.VariantType.OUTLINED,
			shadowNumber: 4,
			isSquare: true
		}

		const wrapper = shallow(
			<Block {...props}>
				<div>mock-children</div>
			</Block>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-block-className',
			variant: Block.VariantType.OUTLINED,
			shadowNumber: 4,
			isSquare: true
		};
		const wrapper = mount(
			<Block {...props}>
				<div>mock-children</div>
			</Block>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().variant).toEqual(props.variant);
		expect(wrapper.props().shadowNumber).toEqual(props.shadowNumber);
		expect(wrapper.props().isSquare).toEqual(props.isSquare);
		expect(wrapper.props().children).toEqual(<div>mock-children</div>);
	})
})