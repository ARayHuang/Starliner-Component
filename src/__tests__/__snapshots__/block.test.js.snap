// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block should render correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="mock-block-className"
  elevation={4}
  square={true}
  variant="outlined"
>
  <div>
    mock-children
  </div>
</WithStyles(ForwardRef(Paper))>
`;
